SplitFiles/slot.c

```diff
@@ -58,48 +60,54 @@ void slot_func(GameComponent* input, GameParameters* params) {
   // Now do the toasting thing
   bool update_required = false;
   bool increment = true;
-  if (state->bread) {
+  if (state->power != state->temperature) {
     update_required = true;
-    breadStateOut = 1;
-    state->power = 200; // Slot is on
-    state->temperature += ((state->power - state->temperature) / 10); //temperature may be rising (or steadyish close to 200)
-    state->bread->temperature += ((state->temperature - state->bread->temperature) / 10); //bread temperature also likely to be rising
+    increment = (state->power > state->temperature);
+    // Update the tick-wise integration of observed temperature differences
+    state->thermalAggregation += (state->power - state->temperature);
+    // Calculate new temperature from that
+    state->temperature = state->thermalAggregation / state->thermalMass;
+#ifdef _TEST_SLOT
+    printf(PRINTAT"\x02\x0B""%d %d %d", state->thermalAggregation, state->temperature, state->power);
+#endif
+  }
+  if (state->bread) {
+    state->bread->thermalAggregation += (state->temperature - state->bread->temperature); //bread temperature also likely to be rising
+    state->bread->temperature = state->bread->thermalAggregation / state->bread->thermalMass;
     if (state->bread->moisture > 0) {
-      //Dry it out before toastuing can commence
-      state->bread->moisture -= (state->bread->temperature / 10);
+      //Dry it out before toasting can commence
+      state->bread->moisture -= MIN(state->bread->moisture, (state->bread->temperature / 10));
```

Comment:
Is this really what you want here? Is it to stop the moisture going negative?

Reply:
Yes; stops it going negative

Reply:
NB updated the initial comment on the PR to better reflect the new simplified scheme (update in italics)